nevergrad/optimization/base.py

```diff
@@ -527,7 +528,13 @@ def minimize(
                 if verbosity and new_sugg:
                     print(f"Launching {new_sugg} jobs with new suggestions")
                 for _ in range(new_sugg):
+                    
                     args = self.ask()
+                    num_tries = 0
+                    while cheap_constraints_checker is not None and cheap_constraints_checker(c) and num_tries < 1000:
+                        self._num_ask -= 1
+                        args = self.ask()
+                        num_tries += 1
```

Comment:
Better loop:
```python
if cheap_constraints_checker is not None:
    for _ in range(1000):
         args = self.ask()
         if cheap_constraints_checker(*args.args, **args.kwargs)
              break
```
Other remarks:
- I don't think you should decrease num `ask`, you **did** ask so it does not really make sense.
- You should also do a `tell`, because many algorithms (I'm mostly thinking of `DE`) will break if you do `ask` without `tell`.

Reply:
Interesting.
It is on purpose that I decrease num_ask. The principle is that constraints are cheap - so we should not pay one budget evaluation when we redraw. No idea of a better design :-(

Reply:
Discussed in chat and solved, thx Jeremy.